### PR TITLE
CL2: fix etcd profiles gathering

### PR DESCRIPTION
**Pull request by skurtzemann**
This PR is to resolve issue #948 as seen with @mm4tt

### CODE REVIEW

**Comment by mm4tt** [comment]
Thanks, @skurtzemann!

Let's see whether it works
/ok-to-test

File: clusterloader2/pkg/measurement/common/profile.go [hunk 1]
```diff
@@ -181,7 +181,15 @@ func (p *profileMeasurement) gatherProfile(c clientset.Interface) ([]measurement
 		return nil, fmt.Errorf("profile gathering failed finding component port: %v", err)
 	}
 	profileProtocol := getProtocolForComponent(p.config.componentName)
-	getCommand := fmt.Sprintf("curl -s -k %slocalhost:%v/debug/pprof/%s", profileProtocol, profilePort, p.config.kind)
+
+	var getCommand string
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/common/profile.go]: Can we extract this logic (constructing getCommand) to a separate method?

File: clusterloader2/pkg/measurement/common/profile.go [hunk 2]
```diff
@@ -181,7 +181,15 @@ func (p *profileMeasurement) gatherProfile(c clientset.Interface) ([]measurement
...
+	if p.config.componentName == "etcd" {
+		etcdCert := "/etc/srv/kubernetes/pki/etcd-apiserver-server.crt"
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/common/profile.go]: Now we confirmed that it works we should create a flags for these two variables and default them to these values. The logic then should be: if flags are set, pass them and use https, if not don't pass and use http

**Comment by mm4tt** [comment]
Works like a charm! Screenshot from the presubmit artifacts:
gJDD9EB3cGM

Ok, now let's clean this up a bit

File: clusterloader2/pkg/measurement/common/profile.go (hunk 1 above)
Comment (skurtzemann) [reply clusterloader2/pkg/measurement/common/profile.go]: Ok, I'll do that

File: clusterloader2/pkg/measurement/common/profile.go (hunk 2 above)
Comment (skurtzemann) [reply clusterloader2/pkg/measurement/common/profile.go]: Use flags or env variables (`ETCD_CERTIFICATE` and `ETCD_KEY`) like https://github.com/kubernetes/perf-tests/blob/39b98208736db08b243b23ff53f1215c022cfb34/clusterloader2/pkg/measurement/common/etcd_metrics.go#L203 ?

Comment (mm4tt) [reply clusterloader2/pkg/measurement/common/profile.go]: Let's use flags. We can use the "env-var" flag for convenience (can be set both via flag and env variable). The advantage is the basically a flag is clearly defined and documented (e.g. I can run --help and check what it does)

Comment (skurtzemann) [reply clusterloader2/pkg/measurement/common/profile.go]: > Now we confirmed that it works we should create a flags for these two variables and default them to these values. The logic then should be: if flags are set, pass them and use https, if not don't pass and use http

I create 2 flags but how can I get theses values into `clusterloader2/pkg/measurement/common/profile.go ` ?
https://github.com/kubernetes/perf-tests/pull/1136/commits/8e492ada85e2038ccebdb88b160ea306da0268da

File: clusterloader2/pkg/measurement/common/profile.go (hunk 1 above)
Comment (skurtzemann) [reply clusterloader2/pkg/measurement/common/profile.go]: Done in https://github.com/kubernetes/perf-tests/pull/1136/commits/b6e71483ad60e548c065d1ed905ba87e101e5809

File: clusterloader2/cmd/clusterloader.go [hunk 3]
```diff
@@ -61,6 +61,8 @@ func initClusterFlags() {
 	flags.StringEnvVar(&clusterLoaderConfig.ClusterConfig.KubeConfigPath, "kubeconfig", "KUBECONFIG", "", "Path to the kubeconfig file")
 	flags.IntEnvVar(&clusterLoaderConfig.ClusterConfig.Nodes, "nodes", "NUM_NODES", 0, "number of nodes")
 	flags.StringEnvVar(&clusterLoaderConfig.ClusterConfig.Provider, "provider", "PROVIDER", "", "Cluster provider")
+	flags.StringEnvVar(&clusterLoaderConfig.ClusterConfig.EtcdCertificatePath, "etcd-certificate", "ETCD_CERTIFICATE", "/etc/srv/kubernetes/pki/etcd-apiserver-server.crt", "Path to ectd certificate")
```
Comment (mm4tt) [inline clusterloader2/cmd/clusterloader.go]: Let's change the description to be clear about where the key/certificate is, e.g. `Path to the etcd certificate on the master machine". Here and below.

File: clusterloader2/pkg/measurement/common/profile.go [hunk 4]
```diff
@@ -221,6 +219,26 @@ func (p *profileMeasurement) shouldExposeApiServerDebugEndpoint() bool {
...
+	profilePort, err := getPortForComponent(p.config.componentName)
+	if err != nil {
+		return "", fmt.Errorf("get profile command failed finding component port: %v", err)
+	}
+	profileProtocol := getProtocolForComponent(p.config.componentName)
+
+	//TODO: use flags for etcdCert and etcdKey
+	var command string
+	if p.config.componentName == "etcd" {
+		etcdCert := "/etc/srv/kubernetes/pki/etcd-apiserver-server.crt"
```
Comment (wojtek-t) [inline clusterloader2/pkg/measurement/common/profile.go]: Let's do this - you already set defaults to those values...

File: clusterloader2/pkg/measurement/common/profile.go [hunk 5]
```diff
@@ -221,6 +219,26 @@ func (p *profileMeasurement) shouldExposeApiServerDebugEndpoint() bool {
 	return p.config.componentName == "kube-apiserver"
 }
 
+func (p *profileMeasurement) getProfileCommand() (string, error) {
+	profilePort, err := getPortForComponent(p.config.componentName)
+	if err != nil {
+		return "", fmt.Errorf("get profile command failed finding component port: %v", err)
```
Comment (mm4tt) [inline clusterloader2/pkg/measurement/common/profile.go]: NIT: Could you use go-errors for all added / updated lines? Here is an example https://github.com/kubernetes/perf-tests/blob/75dfad9672dfabe11b292c246f8f99c79db84655/clusterloader2/pkg/config/template_provider.go#L240

File: clusterloader2/pkg/measurement/common/profile.go (hunk 4 above)
Comment (mm4tt) [inline clusterloader2/pkg/measurement/common/profile.go]: The `Execute` method has the MeasurementConfig argument. You can access the ClusterConfig from there (via ClusterLoaderConfig, IIRC). Then you can either pipe the values you need here, or add them to the profileMeasurement struct.

Comment (wojtek-t) [reply clusterloader2/pkg/measurement/common/profile.go]: @skurtzemann - `start()` method already has access to the config, and that is calling `gatherProfile` - let's just pipe them here

Comment (skurtzemann) [reply clusterloader2/pkg/measurement/common/profile.go]: Ok, thanks for your help. I submit the updated PR with the config

**Comment by skurtzemann** [comment]
If tests succeded I will clean all my commits (rebase from master make my very dirty commits :( )

**Comment by mm4tt** [comment]
Looks good! 
One more thing before I approve, could you please squash all commits into one? 
Thanks!

**Comment by mm4tt** [comment]
If I'm not available, you can also ping @wojtek-t for the approval

**Comment by skurtzemann** [comment]
> Looks good!
> One more thing before I approve, could you please squash all commits into one?
> Thanks!

Commits squashed, tests in progress :)

**Comment by mm4tt** [comment]
Nice!

/lgtm
/approve

**Pull request merged**